packages/stable/src/api/records/types.ts

```diff
@@ -0,0 +1,205 @@
+// Copyright 2025 Cognite AS
+
+/**
+ * @pattern ^[a-zA-Z][a-zA-Z0-9_-]{0,41}[a-zA-Z0-9]?$
+ */
+export type RecordSpaceId = string;
+
+/**
+ * @pattern ^[^\\x00]{1,256}$
+ */
+export type RecordExternalId = string;
+
+/**
+ * A value matching the data type of the defined property
+ * */
+export type RawPropertyValueV3 =
+  | string
+  | number
+  | boolean
+  | object
+  | string[]
+  | boolean[]
+  | number[]
+  | object[];
+
+/**
+ * Group of property values indexed by a local unique identifier. The identifier has to have a length of between 1 and 255 characters.  It must also match the pattern ```^[a-zA-Z0-9][a-zA-Z0-9_-]{0,253}[a-zA-Z0-9]?$``` , and it cannot be any of the following reserved identifiers: ```space```, ```externalId```, ```createdTime```, ```lastUpdatedTime```, ```deletedTime```, and ```extensions```. The maximum number of properties depends on your subscription, and is by default 100.
+ * @example {"someStringProperty":"someStringValue","someDirectRelation":{"space":"mySpace","externalId":"someNode"},"someIntArrayProperty":[1,2,3,4]}
+ */
+export type PropertyValueGroupV3 = Record<string, RawPropertyValueV3>;
+
+/**
+ * Reference to an existing container
+ */
+export interface ContainerReference {
+  /** External-id of the container */
+  externalId: RecordExternalId;
+  /** Id of the space hosting (containing) the container */
+  space: RecordSpaceId;
+  type: 'container';
+}
+
+/**
+ * Sort direction for record queries
+ */
+export type RecordSortDirection = 'ascending' | 'descending';
+
+/**
+ * Timestamp bound for time range filters (ISO 8601 string or Unix ms since epoch)
```

Comment:
Nice, NB in the API docs it is not clear what the expected input format is for string, could use another look 🙂